src/mr/default_memory_resource.cpp

```diff
@@ -14,23 +14,46 @@
  * limitations under the License.
  */
 #include <rmm/mr/cuda_memory_resource.hpp>
+#include <rmm/mr/cnmem_memory_resource.hpp>
+#include <rmm/mr/managed_memory_resource.hpp>
 #include <rmm/mr/default_memory_resource.hpp>
 #include <rmm/mr/device_memory_resource.hpp>
 
 #include <atomic>
 
 namespace rmm {
 namespace mr {
-namespace {
-// The initial, default memory resource is a `cuda_memory_resource`
+
+namespace detail{
 device_memory_resource* initial_resource() {
+  return cuda_resource();
+}
+
+device_memory_resource* cuda_resource() {
   static cuda_memory_resource resource{};
   return &resource;
 }
 
+device_memory_resource* pool_resource(std::size_t pool_size){
+   static cnmem_memory_resource mr{pool_size};
+   return &mr;
+}
+
+
+device_memory_resource* managed_resource(){
```

Comment:
Need to add an implementation for `managed_pool_resource()` as well.

Reply:
yup sorry  I had forgotten to save this file.